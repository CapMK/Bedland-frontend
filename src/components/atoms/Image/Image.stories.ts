import type { Meta, StoryObj } from '@storybook/react';
import DefaultImage from 'assets/img/Loginlogo.svg';
import Image from './Image';

const meta = {
  title: 'Atoms/Image',
  component: Image,
  tags: ['autodocs'],
} satisfies Meta<typeof Image>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    src: DefaultImage,
    alt: '',
  },
};
