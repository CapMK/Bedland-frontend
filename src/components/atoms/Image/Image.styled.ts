import styled from 'styled-components';

export const Img = styled.img`
  display: flex;
  height: 17.625rem;
  width: 22.5rem;
`;

export default Img;
