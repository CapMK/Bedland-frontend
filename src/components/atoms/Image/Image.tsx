// this default image should be changed when the graphic is done
import DefaultImage from 'assets/img/Loginlogo.svg';
import * as Styled from './Image.styled';

type ImageProps = {
  src: string;
  alt: string;
};

const Image = ({ src = DefaultImage, alt = '' }: ImageProps) => <Styled.Img src={src} alt={alt} />;

export default Image;
