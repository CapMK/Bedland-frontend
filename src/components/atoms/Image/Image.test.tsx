import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import DefaultImage from 'assets/img/Loginlogo.svg';

import Image from './Image';

describe('Image', () => {
  afterEach(cleanup);
  it('snapshot renders correctly', () => {
    const snapshot = renderer.create(<Image src={DefaultImage} alt='image' />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
