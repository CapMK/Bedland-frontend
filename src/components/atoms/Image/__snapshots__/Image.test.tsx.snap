// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Image > snapshot renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 17.625rem;
  width: 22.5rem;
}

<img
  alt="image"
  className="c0"
  src="/src/assets/img/Loginlogo.svg"
/>
`;
